rosetta/retriever/retriever.go

```diff
@@ -131,35 +91,64 @@ func (r *Retriever) Current() (identifier.Block, time.Time, error) {
 	return block, header.Timestamp, nil
 }
 
-func (r *Retriever) Balances(network identifier.Network, block identifier.Block, account identifier.Account, currencies []identifier.Currency) ([]object.Amount, error) {
+func (r *Retriever) Balances(block identifier.Block, account identifier.Account, currencies []identifier.Currency) (identifier.Block, []rosetta.Amount, error) {
+
+	// Run validation on the block ID. This also fills in missing information.
+	completeBlock, err := r.validate.Block(block)
+	if err != nil {
+		return identifier.Block{}, nil, fmt.Errorf("could not validate block: %w", err)
```

Comment:
```suggestion
		return identifier.Block{}, nil, fmt.Errorf("invalid block: %w", err)
```

`could not XXX` implies that it's our fault and not that we just received an invalid input IMO. WDYT?

Applies throughout this file

Reply:
The error on validation of the block identifier could indeed be because we cannot validate it (i.e. we were not able to get the header from storage, we were not able to get first or last index block height, etc). So `invalid block` would be misleading.